include/boost/fusion/container/set/set.hpp

```diff
@@ -97,39 +97,20 @@ namespace boost { namespace fusion
 
         template <typename Sequence>
         BOOST_FUSION_GPU_ENABLED
-        set(Sequence const& rhs)
-            : data(rhs) {}
-
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
-        template <typename Sequence>
-        BOOST_FUSION_GPU_ENABLED
         set(Sequence&& rhs)
             : data(std::forward<Sequence>(rhs)) {}
-#endif
 
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(typename detail::call_param<T>::type ...args)
             : data(args...) {}
 
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
         template <typename ...U>
         BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(U&& ...args)
```

Comment:
> The implicit conversion constructor does not currently have the constexpr label. Is that intentional?

Yes. During constructing (or copying, moving, ...) nested container, outer container's constexpr ctor tries to instantiate inner container's ctor including implicit conversion ctor whcih has SFINAE, like `enable_if<is_convertible<U0, T0>>::type`, even if such ctor is not eligible.

[#11267](https://svn.boost.org/trac/boost/ticket/11267) and #79 were affected with this.

> That constructor (template<typename Sequence> set(Sequence&&)) should likely be marked explicit if SFINAE cannot be used for some reason.

Ah, I might mislead you. I meant,

- *implicit conversion* ctor
```cpp
SEQUENCE<int> src;
OTHER_SEQUENCE<int> dst = src; // *explicit ctor* will reject this line
// or 
OTHER_SEQUENCE<int> dst{src};
```
- *explicit conversion*
```cpp
SEQUENCE<int> src;
OTHER_SEQUENCE<int> dst = as_OTHER_SEQUENCE(src);
// or
OTHER_SEQUENCE<int> dst = convert<OTHER_SEQUENCE_tag>(src); // Oh, I realized that there is no documents for fusion::convert!
```